### DISCUSSION TITLE
Use case: nested structure

### DISCUSSION THREAD

**Discussion by lgiordani** [grow/grow#1197] [General]
Hi everybody!

I recently started using Grow, and I like the system a lot. However, I tried to implement a specific use case and not everything went smooth. Some solutions felt convoluted and I couldn't implement all requirements.

This might be because I do not know how to use the system or didn't understand some concepts, or it might be because of limitations of the system or even because of bugs.

I'd like to see the project implemented (this was my original goal, I will use it for my teaching activity), but I also think it might be a good contribution to the documentation once I ironed out the issues I have, so I created a repository with a description of the requirements and three stages of the development, in three different Grow sites.

You can find the repository here https://github.com/lgiordani/grow-test-site.
I hope the README is clear enough, I'd like to use the repository to have a discussion with Grow developers and other users, so feel free to send PRs if you have fixes or better solutions to the problems I highlighted.

Thanks in advance!

Leo

**Comment by jeremydw** [comment]
Hi - Thanks for the extensive sample. One other route you can pursue is adding an extension to create a data structure that would best serve your purposes. For example, you could store the documents as you see fit, and call a function like:

```
{% set module_page = get_module_page(doc) %}
```

Where `get_module_page` is defined in your extension code. https://grow.dev/extensions/create/

**Comment by lgiordani** [comment]
Hi @jeremydw, thanks for the reply. Which issue is this addressing among the ones I mentioned in the sample? Thanks!
